selfservice/hook/web_hook.go

```diff
@@ -307,9 +350,58 @@ func doHttpCall(method string, url string, as AuthStrategy, body io.Reader) erro
 
 	if err != nil {
 		return err
-	} else if resp.StatusCode >= 400 {
+	} else if resp.StatusCode >= http.StatusBadRequest {
+		if interrupt {
+			if err := parseResponse(resp); err != nil {
+				return err
+			}
+		}
 		return fmt.Errorf("web hook failed with status code %v", resp.StatusCode)
 	}
 
 	return nil
 }
+
+func parseResponse(resp *http.Response) (err error) {
+	if resp == nil {
+		return fmt.Errorf("empty response provided from the webhook")
+	}
+
+	body, err := io.ReadAll(resp.Body)
```

Comment:
Instead of reading the body into memory and then unmarshalling, it is much more effective to do:

```
if err := json.NewDecoder(resp.Body).Decode(); err != nil {
```

Reply:
That is neat! Thanks for the tip!  Fix in: 81106d15.